app/services/foreman/render_templates_from_folder.rb

```diff
@@ -0,0 +1,143 @@
+    end
+
+    def render_template(template)
+      source = source_klass.new(template)
+      scope = Foreman::Renderer.get_scope(host: host(template), source: source)
+      Foreman::Renderer.render(source, scope)
+    end
+
+    def templates
+      @templates ||= files.map { |filepath| load_file(filepath) }
+    end
+
+    def snippets
+      templates.select { |template| template.snippet? }
+    end
+
+    def load_file(filepath)
+      content = File.read(filepath)
+      metadata = Template.parse_metadata(content).deep_symbolize_keys
+      name = metadata[:name]
+      raise "Could not read name from metadata in template #{filepath}." unless name.present?
+      kind = metadata[:kind]
+      snippet = (kind == 'snippet') || !!metadata[:snippet]
+      oses = metadata[:oses] || ['RedHat']
+      operatingsystems = oses.map { |os| build_operatingsystem(name: os) }.compact
+      ProvisioningTemplate.new(name: name, template: content, template_kind: TemplateKind.new(name: kind), snippet: snippet, operatingsystems: operatingsystems, vendor: path)
+    end
+
+    def host(template)
+      operatingsystem = template.operatingsystems.first
+      build_host(operatingsystem: operatingsystem)
+    end
+
+    private
+
+    def fetch_metadata(content, key, default = nil)
+      content.split("\n").find { |l| l.starts_with?("#{key}: ") }.try(:remove, "#{key}: ").try(:strip) || default
+    end
+
+    def source_klass
+      Foreman::Renderer::Source::Directory
+    end
+
+    def files
+      @files ||= Dir.glob(File.join(path, '**', '*.erb'))
+    end
+
+    def architecture
+      @architecture ||= build_architecture
+    end
+
+    def build_architecture
+      Architecture.new(
+        name: 'x86_64'
+      )
+    end
+
+    def build_operatingsystem(name:)
+      family = Operatingsystem.deduce_family(name)
+      return unless family
+      family.constantize.new(
+        name: name,
+        major: 7,
+        minor: 6,
+        family: family,
+        architectures: [architecture]
+      )
+    end
+
+    def build_medium(operatingsystem:)
+      case operatingsystem.family
+      when 'Solaris'
+        Medium.new(
+          name: 'Solaris Medium',
+          path: 'http://www.example.com/vol/solgi_5.10/sol$minor_$release_$arch',
+          media_path: 'www.example.com:/vol/solgi_5.10/sol$minor_$release_$arch',
+          config_path: 'www.example.com:/vol/jumpstart',
+          image_path: 'www.example.com:/vol/solgi_5.10/sol$minor_$release_$arch/flash/',
+          os_family: operatingsystem.family
+        )
+      else
+        Medium.new(
+          name: 'Example Medium',
+          path: 'http://www.example.com/path/',
+          os_family: operatingsystem.family
+        )
+      end
+    end
+
+    def build_host(operatingsystem:)
+      medium = build_medium(operatingsystem: operatingsystem)
+      domain = Domain.new(
+        name: 'example.com'
+      )
+      subnet = Subnet.new(
+        name: 'Example Subnet',
```

Comment:
We could use the RFC5735 name for it:
```suggestion
        name: 'TEST-NET-1',
```